test/torchaudio_unittest/rnnt/autograd_impl.py

```diff
@@ -0,0 +1,98 @@
+from typing import Callable, Tuple
+import torch
+from torch import Tensor
+from torch.autograd import gradcheck
+from torchaudio_unittest.common_utils import (
+    TestBaseMixin,
+)
+from torchaudio.prototype.rnnt_loss import RNNTLoss, rnnt_loss
+from parameterized import parameterized
+from .utils import (
+    numpy_to_torch,
+    get_B1_T10_U3_D4_data,
+    get_numpy_data_B2_T4_U3_D3,
+    get_numpy_data_B1_T2_U3_D5
+)
+from .numpy_transducer import NumpyTransducerLoss
+
+
+class Autograd(TestBaseMixin):
+    @staticmethod
+    def get_data(data_func, device):
+        data_np = data_func()
+        if type(data_np) == tuple:
+            data_np = data_np[0]
+        data = numpy_to_torch(
+            data=data_np, device=device, requires_grad=True
+        )
+        return data
+
+    def assert_grad(
+            self,
+            loss: Callable[..., Tensor],
+            inputs: Tuple[torch.Tensor],
+            *,
+            enable_all_grad: bool = True,
+    ):
+        inputs_ = []
+        for i in inputs:
+            if torch.is_tensor(i):
+                i = i.to(dtype=self.dtype, device=self.device)
+                if enable_all_grad:
+                    i.requires_grad = True
+            inputs_.append(i)
+        assert gradcheck(loss, inputs, eps=1e-03, atol=1e-03, rtol=1e-03, nondet_tol=0.)
```

Comment:
can you check if atol can be reduced any further?

Reply:
This needs to be documented in the test code. This looks much lower precision than other autograd tests we have in torchaudio.

Reply:
unfortunately, that's the lowest for atol and eps as expected for float32. rtol can be reduced to 0, but this seems too good, so let's use the default value ([0.001](https://pytorch.org/docs/stable/autograd.html#torch.autograd.gradcheck)).

Reply:
> This needs to be documented in the test code. This looks much lower precision than other autograd tests we have in torchaudio.

As commented [here](https://github.com/carolineechen/audio/pull/2#discussion_r631342724) this is due to the use of float32. added a comment above the line.